src/web_buffer.h

```diff
@@ -1,77 +1,266 @@
 #ifndef NETDATA_WEB_BUFFER_H
 #define NETDATA_WEB_BUFFER_H 1
 
-#define WEB_DATA_LENGTH_INCREASE_STEP 1024
+/**
+ * @file web_buffer.h
+ * @brief Data structure and API of web_buffer.
+ *
+ * A web buffer is a buffer to store data for sending or recieving on web clients or servers.
+ * A web buffer increases it's size automatic.
+ * A web buffer is always null terminated.
+ */
 
+#define WEB_DATA_LENGTH_INCREASE_STEP 1024 ///< Minimum size a web buffer gets increased.
+
+/** web buffer */
 typedef struct web_buffer {
-    size_t size;        	// allocation size of buffer, in bytes
-    size_t len;     		// current data length in buffer, in bytes
-    char *buffer;   		// the buffer itself
-    uint8_t contenttype;	// the content type of the data in the buffer
-    uint8_t options;		// options related to the content
-    time_t date;    		// the timestamp this content has been generated
-    time_t expires;			// the timestamp this content expires
+    size_t size;        	///< Total size of `buffer`, in bytes.
+    size_t len;     		///< Used size of `buffer`, in bytes.
+    char *buffer;   		///< The conetent
+    uint8_t contenttype;	///< Content type of the buffer. (CT_*)
+    uint8_t options;		///< WB_CONTENT_*
+    time_t date;    		///< Timestamp the content has been created.
+    time_t expires;			///< Timestamp the content expires.
 } BUFFER;
 
 // options
-#define WB_CONTENT_CACHEABLE            1
-#define WB_CONTENT_NO_CACHEABLE         2
+#define WB_CONTENT_CACHEABLE            1 ///< ktsaou: Your help needed
```

Comment:
`WB_CONTENT_CACHEABLE` instructs the web server to respond with HTTP headers stating that the response can be cached by proxy servers and web clients for some time (currently fixed to 1 day).